src/handlers/graphical.rs

```diff
@@ -688,6 +697,30 @@ impl GraphicalReportHandler {
         Ok(())
     }
 
+    fn render_highlighted_line_text(
+        &self,
+        f: &mut impl fmt::Write,
+        styled_text: &[Styled<&str>],
+    ) -> fmt::Result {
+        for styled in styled_text {
+            let styled_text = styled.to_string();
+            for (c, width) in styled_text
+                .chars()
+                .zip(self.line_visual_char_width(&styled_text))
```

Comment:
@Benjamin-L  yeah. several options I see here:

1) what you suggested, have a `starting_column` parameter
2) change `line_visual_char_width` to work over `Vec<Styled<&str>>` and then adjust other calls to work with that
3) `to_string()` the entire line prior to passing into `render_line_text`, but we need to make sure that all formatting logic correctly handled the presence of zero-width escape codes. It seems like this might not be a problem since the `width` output is only considered for `\t` characters.

What do you think about option 3? I think it might be the simplest approach, but I'm unsure what will happen by introducing zero-width escape sequences.

Reply:
I don't think there would be any impacts with option 3, but there is a latent possibility for future bugs because `line_visual_char_width` will produce incorrect widths for escape sequences.

Reply:
For option 3, the width output is only used for rendering `\t`, but the width calculation for tabs depends on the width of the previous characters. In order to make this work we would have to modify `line_visual_char_width` to calculate ANSI escape widths correctly. This wouldn't be too bad, could probably just have it enter a zero-width state when it sees `\x1b` and then go back to normal when it sees `m`.

Thinking about this more carefully, I bet this issue is affecting us already, since we're passing `styled.to_string()` to `line_visual_char_width` for each fragment. We're going to have to fix it either way, so I'd just do that and then go for option 3.

I wouldn't worry too much about the allocations, miette's formatting logic is already full of them, and isn't particularly perf-sensitive.

Reply:
I'm mildly regretting introducing the tab and widechar support because of how much of a headache it's being with everything else. Worth it for not having my nushell errors look completely busted, but oh boy is this a lot of incidental complexity :)

Reply:
Fixed this in 5957da3bbe10adb9e92183f11aec0ff14ba87f1b